src/Cli/func/Actions/LocalActions/PackAction/CustomPackSubcommandAction.cs

```diff
@@ -19,6 +22,80 @@ public async Task RunAsync(PackOptions packOptions)
+                hostJsonExists,
+                hostJsonExists ? null : "Required file 'host.json' not found. Ensure this is a valid Azure Functions project.");
+
+            if (!hostJsonExists)
+            {
+                PackValidationHelper.DisplayValidationEnd();
+                throw new CliException($"Required file 'host.json' not found in {functionAppRoot}. Custom handler function apps require host.json.");
+            }
+
+            // Validate custom handler configuration and executable
+            try
+            {
+                var hostJsonContent = FileSystemHelpers.ReadAllTextFromFileAsync(hostJsonPath).Result;
+                var hostConfig = JObject.Parse(hostJsonContent);
+
+                var customHandler = hostConfig["customHandler"];
+                if (customHandler != null)
+                {
+                    var description = customHandler["description"];
+                    if (description != null)
+                    {
+                        var defaultExecutablePath = description["defaultExecutablePath"]?.ToString();
+                        if (!string.IsNullOrEmpty(defaultExecutablePath))
+                        {
+                            var executablePath = Path.Combine(functionAppRoot, defaultExecutablePath);
+                            var executableExists = FileSystemHelpers.FileExists(executablePath);
+
+                            if (executableExists)
+                            {
+                                PackValidationHelper.DisplayValidationResult("Validate Custom Handler Executable", true);
+                            }
+                            else
+                            {
+                                PackValidationHelper.DisplayValidationWarning(
+                                    "Validate Custom Handler Executable",
+                                    $"Custom handler executable '{defaultExecutablePath}' not found. Ensure the executable exists or will be provided by other deployment methods.");
+                            }
+                        }
+                        else
+                        {
+                            PackValidationHelper.DisplayValidationResult("Validate Custom Handler Configuration", true, "No defaultExecutablePath specified in host.json");
+                        }
+                    }
+                    else
+                    {
+                        PackValidationHelper.DisplayValidationResult("Validate Custom Handler Configuration", true, "Custom handler configuration found but no description specified");
+                    }
+                }
+                else
+                {
+                    PackValidationHelper.DisplayValidationWarning(
+                        "Validate Custom Handler Configuration",
+                        "No custom handler configuration found in host.json. This may be intentional if configuration is provided by other means.");
```

Comment:
I would suggest keeping this clean and just have the first part. Second sentence might confuse the customers. If we want we can add an aka.ms link that points to the documentation that explains other ways of configuring `host.json` or provide a way to suppress this warning.